storage/innobase/handler/ha_innodb.cc

```diff
@@ -6755,13 +6756,17 @@ wsrep_store_key_val_for_row(
 			/* cannot exceed max column lenght either, we may need to truncate
 			the stored value: */
 			if (true_len > sizeof(sorted)) {
-			  true_len = sizeof(sorted);
+				true_len = sizeof(sorted);
 			}
 
-			memcpy(sorted, data, true_len);
-			true_len = wsrep_innobase_mysql_sort(
-				mysql_type, cs->number, sorted, true_len,
-				REC_VERSION_56_MAX_INDEX_COL_LEN);
+			ut_ad(data || (data == nullptr && true_len == 0));
+			if (data) {
+				memcpy(sorted, data, true_len);
```

Comment:
It would be more optimal to change the condition to `if (true_len)`. This is because in a `cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo` build that variable would likely already reside in the return value register, such as register 0 on ARM or `eax` on i686 or `rax` on x86-64, after the preceding call `true_len = (ulint) my_well_formed_length(…);`.

I would suggest to write this condition and the preceding assertion as follows:
```c++
if (true_len) {
	true_len = wsrep_innobase_mysql_sort(…);
} else {
	ut_ad(!data);
}
```

Reply:
Fixed.